content/en/docs/tasks/security/authentication/authn-policy/authn-policy_snip.txt

```diff
@@ -0,0 +1,451 @@
+####################################################################################################
+# WARNING: THIS IS AN AUTO-GENERATED FILE, DO NOT EDIT. PLEASE MODIFY THE ORIGINAL MARKDOWN FILE
+####################################################################################################
+
+snip_line_23() {
```

Comment:
What if a doc gets an extra line? Does this break the test the next time the generator is run? Also, I feel like referring to everything as line_* might make it hard to maintain tests. I know we have the option of `snip_id` so the decision not to rely on the auto-generated name might rely on the test writer/reviewer.

Reply:
Good point. Any change to an md file will trigger a regen of its snippet file, which may cause the test to fail. If using the default snippet names, any change that adds or removes lines will likely break the test, even if the snippet contents hasn't changed at all. So, using snip_ids should definitely be a recommended best practice. Only actual command changes will then break the test.

Reply:
A serialized default could also be useful. You can suffix an index instead of the line number and that would solve the issue. If commands change order, the test should be revisited, so that's shouldn't be an issue. We should avoid requiring manual suffix IDs for every snippet, IMO. That way we avoid making authoring content unnecessarily cumbersome.

Reply:
@rcaballeromx Do you mean something like `snip_<section heading>_<text section number>`?
For traffic-shifting.md, it would look something like this:
```
snip_apply_weight_based_routing_1()
snip_apply_weight_based_routing_2()
...
snip_cleanup_1()
```
That sounds reasonable to me. Still could break, but a lot less fragile than line numbers.

Reply:
Yes, although maybe using the folder name will help keep them short and readable.
`snip_authn-policy_1()` for example.

Reply:
My suggestion is less fragile. Only the following snippets in the same section of the doc will break (change) if one is added or removed. With your version, all the following snippets will break.